scenario/constraints/ReservesConstraint.ts

```diff
@@ -0,0 +1,64 @@
+import { Constraint, Solution } from '../../plugins/scenario';
+import { CometContext } from '../context/CometContext';
+import { expect } from 'chai';
+import { Requirements } from './Requirements';
+import { exp } from '../../test/helpers';
+import { ComparisonOp, parseAmount, getToTransferAmount } from '../utils';
+
+export class ReservesConstraint<T extends CometContext, R extends Requirements> implements Constraint<T, R> {
+  async solve(requirements: R, initialContext: T) {
+    const reservesRequirement = requirements.reserves;
+    if (reservesRequirement !== undefined) {
+
+      const comet = await initialContext.getComet();
```

Comment:
I think we should be pulling Comet from the context in the `barelyMeet` function instead.

Reply:
updated